Fix loading of filters

cammellos opened the pull request:
Currently on some devices there are still some legacy chats(?) that are
one-to-one but don't have a public key as an id (transactor,demo-bot).
We were wrongly sending those to status-go to create filters.

Given the impact of the change and the fact that I don't know how to replicate it, it would be a skip QA.

status: ready

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | b07b1aa5 | [#1](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-8671/1/) | 2019-07-30 11:37:53 | ~33 sec | `android-e2e` | [:page_facing_up: log](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-8671/1/consoleText) |

cammellos committed:
Fix loading of filters

Currently on some devices there are still some legacy chats(?) that are
one-to-one but don't have a public key as an id (transactor,demo-bot).
We were wrongly sending those to status-go to create filters.

Signed-off-by: Andrea Maria Piana <andrea.maria.piana@gmail.com>
